impexp-kml-collada-plugin/src/main/java/org/citydb/modules/kml/controller/KmlExporter.java

```diff
@@ -305,29 +247,29 @@ else if (!Files.isExecutable(collada2gltf))
 
 		// tiling
 		Tiling tiling = query.getTiling();
-		KmlTilingOptions tilingOptions = null;
-		Predicate predicate = null;
 		boolean useTiling = query.isSetTiling();
-		int remainingTiles = 1;
-		int rows = useTiling ? tiling.getRows() : 1;  
+		int rows = useTiling ? tiling.getRows() : 1;
 		int columns = useTiling ? tiling.getColumns() : 1;
-
-		if (useTiling) {
+		if (!useTiling) {
 			try {
-				// transform tiling extent to WGS84
-				tiling.transformExtent(DatabaseConfig.PREDEFINED_SRS.get(DatabaseConfig.PredefinedSrsName.WGS84_2D), databaseAdapter);
-				tilingOptions = tiling.getTilingOptions() instanceof KmlTilingOptions ? (KmlTilingOptions)tiling.getTilingOptions() : new KmlTilingOptions();
-				predicate = query.isSetSelection() ? query.getSelection().getPredicate() : null;
-
-				// calculate and display number of tiles to be exported
-				int displayFormats = config.getKmlExportConfig().getDisplayForms().getActiveDisplayFormsAmount();
-				remainingTiles = rows * columns * displayFormats;
-				log.info(remainingTiles + " (" + rows + "x" + columns + "x" + displayFormats + ") tiles will be generated.");	
+				tiling = new Tiling(config.getKmlExportConfig().getQuery().getSpatialFilter().getExtent(), rows, columns);
```

Comment:
I am not sure, but it seems that this line assumes that the spatial extent was automatically calculated in the `ConfigQueryBuilder`. This would explain why the spatial extent is calculated there even if it is not used (see my comment above).

If this is true, I recommend to resole this "side effect dependency". I suggest that only the `KMLExporter` controller should check whether or not a bbox must be automatically calculated based on the provided query config. And it should trigger the calculation of the bbox, and adapt the query config afterwards if needed. This (possibly adapted) query config should then be passed to the `ConfigQueryBuilder`, and the `ConfigQueryBuilder` should just map it to `Query` object (no further logic).

Reply:
You are right, line 255 assumes that the spatial extent was automatically calculated. However, it is indeed used in the output master and JSON files.

Reply:
Yes, understood. Testing and automatically calculating the bbox should better be done in `KMLExporter` and not in `ConfigQueryBuilder` in my opinion. This would be clearer and avoid the side effect.

Reply:
Good idea. Done in 4327dbc